src/service/storage_arango.py

```diff
@@ -0,0 +1,103 @@
+"""
+A storage system for collections based on an Arango backend.
+"""
+
+from aioarango.database import StandardDatabase
+from aioarango.exceptions import CollectionCreateError
+from src.service.models import SavedCollection
+
+
+# service collection names that aren't shared with data loaders.
+_COLL_PREFIX = "kbcoll_coll_"
+_COLL_COUNTERS = _COLL_PREFIX + "counters"
+_COLL_VERSIONS = _COLL_PREFIX + "versions"
+_COLL_ACTIVE = _COLL_PREFIX + "active"
+
+_FLD_KEY = "_key"
+_FLD_COLLECTION = "collection"
+_FLD_COLLECTION_ID = "collection_id"
+_FLD_COUNTER = "counter"
+
+_QUERY_GET_NEXT_VERSION = f"""
+    UPSERT {{{_FLD_KEY}: @{_FLD_COLLECTION_ID}}}
+        INSERT {{{_FLD_KEY}: @{_FLD_COLLECTION_ID}, {_FLD_COUNTER}: 1}}
+        UPDATE {{{_FLD_COUNTER}: OLD.{_FLD_COUNTER} + 1}}
+        IN @@{_FLD_COLLECTION}
+        OPTIONS {{exclusive: true}}
+        RETURN NEW
+"""
+
+# TODO SCHEMA may want a schema checker at some point... YAGNI for now. See sample service
+
+# Assume here that we're never going to make an alternate implementation of this interface.
+# Seems incredibly unlikely
+
+async def create_storage(
+        db: StandardDatabase,
+        create_collections_on_startup: bool = False):
+    """
+    Create the wrapper.
+
+    db - the database where the data is stored. The DB must exist.
+    create_collections_on_startup - on starting the wrapper, create all collections and indexes
+        rather than just checking for their existence. Usually this should be false to
+        allow for system administrators to set up sharding to their liking, but auto
+        creation is useful for quickly standing up a test service.
+    """
+    if create_collections_on_startup:
+        # TODO DB create indexes and collections
+        await _create_collection(db, _COLL_COUNTERS)  # no indexes necessary
```

Comment:
should `_create_collection` and `_check_collection_exists` take a list of default collections? or `kbcoll_coll_counters` is the only collection we care about?

Reply:
I'll add collections as needed. Currently I'm just adding a method call for each collection since there's only 3 total in my WIP at this point. We could change it to a list later if needed.

Reply:
👍